_scripts/update.py

```diff
@@ -193,5 +222,22 @@ def write_commit_msg(args, new_files: int, updated_files: int):
     )
     args = parser.parse_args()
 
-    update = update_str(cfg_automation, args.branch, args.firefox, args.configs)
-    write_commit_msg(args, *update)
+    paths: list[str] = list()
+    update = update_str(cfg_automation, args.branch, args.firefox, args.configs, paths)
+
+    new_path_msg: str = ""
+    if cfg_automation["paths"] != paths and args.branch == cfg_automation["head"]:
+        removed = list(set(cfg_automation["paths"]) - set(paths))
+        added = list(set(paths) - set(cfg_automation["paths"]))
+        new_path_msg = "\nUpdated paths for sparse checkout: \n"
+        new_path_msg += (
+            f"- {len(removed)} removed ({', '.join(removed)})" if removed else ""
+        )
+        new_path_msg += f"- {len(added)} added ({','.join(added)})" if added else ""
+
+        # Write back updated configuration
+        cfg_automation["paths"] = paths
+        with open(config_file, "w") as f:
+            f.write(json.dumps(cfg_automation, indent=2, sort_keys=True))
```

Comment:
This is the wrong place for this code. The summary data should be generated in `update_str` (which should be renamed as just `update` given this scope creep) and the commit message string in `write_commit_msg`. We also do not need to replicate the itemised changes in the commit message, that's in the actual `config.json` changes.

If there is a real need for a commit message body, please make sure to separate it from the title line by an empty line.

Reply:
After a first try, I dropped completely the idea of updating the commit message, not a lot of value also considering the frequency.